pkg/tests/upgrade_tests/upgrade_test.go

```diff
@@ -635,3 +635,213 @@ func copyMetrics(metrics []prompb.TimeSeries) []prompb.TimeSeries {
 	}
 	return out
 }
+
+func TestPrereleaseExtensionUpgrade(t *testing.T) {
+	var err error
+	var version string
+	ctx := context.Background()
+
+	db, dbContainer, closer := startDB(t, ctx)
+	defer closer.Close()
+
+	defer testhelpers.StopContainer(ctx, dbContainer, false)
+
+	// start promscale & test upgrade-prerelease-extensions as false
+	func() {
+		connectorImage := "vineeth97/promscale:t1"
+		databaseName := "postgres"
+		flags := []string{"-upgrade-prerelease-extensions", "false"}
+		connector, err := testhelpers.StartConnectorWithImage(ctx, connectorImage, *printLogs, flags, databaseName)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+		defer testhelpers.StopContainer(ctx, connector, *printLogs)
+
+		err = db.QueryRow(ctx, `SELECT extversion FROM pg_extension where extname='timescaledb'`).Scan(&version)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		if version != "2.0.0-rc4" {
+			t.Fatal("failed to verify upgrade extension with -upgrade-prerelease-extension false")
+		}
+	}()
+
+	extVersion := "2.0.0-rc2"
+	dropAndCreateExt(t, ctx, extVersion)
+
+	db, err = pgx.Connect(ctx, testhelpers.PgConnectURL("postgres", testhelpers.Superuser))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = db.QueryRow(ctx, `SELECT extversion FROM pg_extension where extname='timescaledb'`).Scan(&version)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	db.Close(ctx)
+	if version != extVersion {
+		t.Fatal("failed to verify upgrade extension with -upgrade-prerelease-extension false")
+	}
+
+	// start a new connector and test --upgrade-prerelease-extensions as true
+	func() {
+
+		connectorImage := "vineeth97/promscale:t1"
+		databaseName := "postgres"
+		flags := []string{"-upgrade-prerelease-extensions", "true"}
+		connector, err := testhelpers.StartConnectorWithImage(ctx, connectorImage, *printLogs, flags, databaseName)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+		defer testhelpers.StopContainer(ctx, connector, *printLogs)
+
+		var version string
+		db, err = pgx.Connect(ctx, testhelpers.PgConnectURL("postgres", testhelpers.Superuser))
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = db.QueryRow(ctx, `SELECT extversion FROM pg_extension where extname='timescaledb'`).Scan(&version)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+		db.Close(ctx)
+		if version != "2.0.0-rc4" {
+			t.Fatal("failed to verify upgrade extension with -upgrade-prerelease-extension true")
+		}
+		t.Logf("successfully tested extension upgrade flow with --upgrade-prereleases-extensions true")
+	}()
+
+	os.Exit(0)
+}
+
+func TestExtensionUpgrade(t *testing.T) {
+	var err error
+	var version string
+	ctx := context.Background()
+
+	db, dbContainer, closer := startDB(t, ctx)
+	defer testhelpers.StopContainer(ctx, dbContainer, false)
+
+	defer closer.Close()
+
+	// start promscale & test upgrade-extensions as false
+	func() {
+		connectorImage := "vineeth97/promscale:t1"
```

Comment:
Currently, I am using image with changes from this branch to make tests succeed but after the next release. I can update to the official promscale latest image.

Reply:
what we should do here is use the tag timescale/promscale:latest and have the ci execute something like https://github.com/timescale/promscale/blob/master/docker-compose/high-availability/test.sh#L11 before this test